lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -2087,13 +2087,54 @@ static NodeValue convertConstant(Module &mod, Constant &constant,
   }
 }
 
+/// Compute number of significant bits that are used to represent data of type
+/// \p kind. For FP, it is the number of bits in mantissa, for integers it's the
+/// number of bits except sign bit.
+/// \p returns number of significant bits of \p kind.
+static size_t numSignificantBits(ElemKind kind) {
```

Comment:
Please add a TODO as discussed to more correctly handle this in the future as we add more ElemKinds.